### PR TITLE
[WIP] Update Ruby cdn_fonts

### PR DESCRIPTION
**Pull request by JLoppert**
### WHY are these changes introduced?

Fixes #0000 

### Measuring impact

How do we know this change was effective? Please choose one:

- [ ] n/a - this doesn't need measurement, e.g. a linting rule or a bug-fix
- [ ] Existing analytics will cater for this addition
- [ ] PR includes analytics changes to measure impact

### Checklist

- [ ] I've considered possible cross-platform impacts (Mac, Linux, Windows)
- [ ] I've considered possible [documentation](https://shopify.dev) changes
- [ ] I've made sure that any changes to `dev` or `deploy` have been reflected in the [internal flowchart](https://www.figma.com/file/7vqUp50u6dm48Zfb4JRRn8/CLI3-Internals).

### CODE REVIEW

File: packages/cli-kit/assets/cli-ruby/lib/shopify_cli/theme/dev_server/cdn_fonts.rb [hunk 1]
```diff
@@ -21,40 +21,45 @@ def call(env)
 
           # Proxy the request, and replace the URLs in the response
           status, headers, body = @app.call(env)
-          body = replace_font_urls(body)
+
+          if headers["Content-Type"]&.start_with?("text/")
```
Comment (JLoppert) [inline packages/cli-kit/assets/cli-ruby/lib/shopify_cli/theme/dev_server/cdn_fonts.rb:25]: Relevant change here.

(Other changes were result of repo/editor configuration)

File: packages/cli-kit/assets/cli-ruby/dev.yml [hunk 2]
```diff
@@ -6,7 +6,7 @@ env:
   SHOPIFY_CLI_DEVELOPMENT: "1"
 
 up:
-  - ruby: 3.1.0
+  - ruby: 3.2.0
```
Comment (JLoppert) [inline packages/cli-kit/assets/cli-ruby/dev.yml:9]: TODO: Verify old behavior with new test passes on 3.1 and fails on 3.2

**Review by yknx4 — COMMENTED** [review]
can we undo the linter changes? it adds some noise

**Comment by JLoppert** [comment]
> can we undo the linter changes? it adds some noise

👍 

I thought it was part of the repo config, but I think it was something that was auto added to my vscode settings.json config:

```json
    "[ruby]": {
        "editor.defaultFormatter": "Shopify.ruby-lsp",
        "editor.formatOnSave": true,
        "editor.tabSize": 2,
        "editor.insertSpaces": true,
        "editor.semanticHighlighting.enabled": true,
        "editor.formatOnType": true
    },
```

I'll comment this config out for the duration of this PR.

**Pull request closed without merging**